src/coreclr/jit/registerarm64.h

```diff
@@ -115,7 +115,7 @@ REGDEF(P12,  12+PBASE, PMASK(12),  "p12",  "na")
 REGDEF(P13,  13+PBASE, PMASK(13),  "p13",  "na")
 REGDEF(P14,  14+PBASE, PMASK(14),  "p14",  "na")
 REGDEF(P15,  15+PBASE, PMASK(15),  "p15",  "na")
-
+REGALIAS(FFR, P14)
```

Comment:
Why use P14 rather than allocate a new P16 (which closely mirrors how thread suspend tracks it)?

Won't this be problematic since P14 is available for general purpose use otherwise?

Reply:
I started with that approach, but soon realized that it gets assigned to related intervals as well causing us to assign a non-existent register. If I do not make it as part of `RBM_ALLMASK`, we still hit various asserts when we try to create RefPosition or assign register that `P16` is non-existent register and is not part of `RBM_ALLMASK`.

Reply:
It's not clear why this would be an issue here, but not for `REG_EBP`/`REG_ESP`/`REG_K0` (x64) or `REG_IP0`/`REG_IP1`/`REG_PR`/`REG_FP`/`REG_ZR` (arm64). These are usable, but come with special semantics.

Is the issue perhaps that FFR needs to be in `ALLMASK` but not part of the candidate set for normal mask nodes?

Reply:
> Is the issue perhaps that FFR needs to be in ALLMASK but not part of the candidate set for normal mask nodes?

exactly

Reply:
Don’t we need to do effectively that here anyways, otherwise LSRA could assign or need pg14 for something else since it is part of the candidate list?

At which point we can just do this without stealing an existing register def away by defining a new pg17?

Reply:
I have already tried that approach. If i have `REG_P16`,  we will allocate that as a register to related intervals too. However, we won't be able to use it directly anyway for storing FFR on stack or for reading the contents out of FFR. To do that, we need additional register and reusing `REG_P14` fits nicely with that mode. `REG_P14` will be busy so it won't get assigned to something else.

Reply:
Registers like the ones you mention are not modelled for spilling at all within LSRA. I do not think it would be that simple to try to model `FFR` as a new register; we end up having to solve what it means to spill and unspill this register, which the JIT only knows how to do to and from stack today. As Kunal mentions that ends up requiring a predicate register anyway, and special spilling logic (much like upper vector saves/restores).

I think the approach taken by this PR (just model the stores/defs to/from a normal mask typed local) is just fine. It means we are essentially manually inserting spills/reloads of the wrffr/rdffr in the IR before LSRA, which means we don't get the benefit of the interference analysis that LSRA has. That part is solvable with a simple analysis that we can do for functions that use FFR. I think that optimization can be done as future work (I don't mind writing that analysis eventually -- ping me when we start implementing BCL methods using these intrinsics...).

Reply:
yes, I was adding the entry of `REG_FFR` in a wrong spot. Fixed it.